code/src/gfx.c

```diff
@@ -142,6 +170,29 @@ static bool IsEntranceDiscovered(s16 index) {
     return isDiscovered;
 }
 
+static bool IsDungeonDiscovered(DungeonId dungeonId) {
+    if (dungeonId <= DUNGEON_GERUDO_TRAINING_GROUNDS) {
+        if (gSettingsContext.dungeonModesKnown) {
+            return true;
+        }

```

Comment:
Is this something that's set if all dungeons are V or MQ, or is it a separate setting?

Reply:
It's set by the app if no dungeon modes will be randomly selected (a mix of MQ and Vanilla can still be all known if the MQs were all added in specifically by the user).